code/modules/projectiles/guns/energy.dm

```diff
@@ -142,46 +145,53 @@
 	return
 
 /obj/item/gun/energy/update_icon()
-	cut_overlays()
+	..(ALL)
+	var/mob/living/carbon/human/user = loc
+	if(istype(user))
+		if(user.hand) //this is kinda ew but whatever
```

Comment:
Could be handled by signals. Have the gun register the user to `update_icon` when it is put in either hand.
Prob out of scope though